src/pages/audio-rewards-page/ChallengeRewardsTile.tsx

```diff
@@ -138,9 +147,22 @@ const RewardsTile = ({ className }: RewardsTileProps) => {
   const dispatch = useDispatch()
   const rewardIds = useRewardIds()
   const userChallengesLoading = useSelector(getUserChallengesLoading)
+  const currentStepCountOverrides = useOptimisticChallengeCompletionStepCounts()
+  const [haveChallengesLoaded, setHaveChallengesLoaded] = useState(false)
 
   useEffect(() => {
-    dispatch(fetchUserChallenges())
+    if (!userChallengesLoading && !haveChallengesLoaded) {
+      setHaveChallengesLoaded(true)
+    }
+  }, [userChallengesLoading, haveChallengesLoaded])
+
+  // poll for user challenges and user balance to refresh
+  useEffect(() => {
+    dispatch(refreshUserChallenges())
```

Comment:
nit: What's the benefit of a separate saga here vs something like `setInterval` and dispatch `getUserChallenges`?

Reply:
Or rather, `useInterval`: https://usehooks-ts.com/react-hook/use-interval